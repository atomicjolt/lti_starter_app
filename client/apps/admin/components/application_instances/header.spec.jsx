import React from 'react';
import TestRenderer from 'react-test-renderer';
import Header from './header';

describe('application instances header', () => {
  let result;
  let instance;
  let props;
  let newInstance;

  beforeEach(() => {
    newInstance = false;
    props = {
      application: {
        name: 'test application',
      },
      newApplicationInstance: () => { newInstance = true; },
    };
    result = TestRenderer.create(<Header {...props} />);
    instance = result.root;
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the onClick function', () => {
    expect(newInstance).toBeFalsy();
    instance.findByType('button').props.onClick();
    expect(newInstance).toBeTruthy();
  });
});
