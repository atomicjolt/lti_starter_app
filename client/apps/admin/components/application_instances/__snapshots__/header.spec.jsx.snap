// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application instances header matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    application={
      Object {
        "name": "test application",
      }
    }
    newApplicationInstance={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="c-title"
        >
          <h1>
            test application
             Instances
          </h1>
        </div>,
        <button
          className="c-btn c-btn--yellow"
          onClick={[Function]}
        >
          New Application Instance
        </button>,
      ],
      "className": "c-info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            test application
             Instances
          </h1>,
          "className": "c-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "test application",
              " Instances",
            ],
          },
          "ref": null,
          "rendered": Array [
            "test application",
            " Instances",
          ],
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "New Application Instance",
          "className": "c-btn c-btn--yellow",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "New Application Instance",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="c-title"
          >
            <h1>
              test application
               Instances
            </h1>
          </div>,
          <button
            className="c-btn c-btn--yellow"
            onClick={[Function]}
          >
            New Application Instance
          </button>,
        ],
        "className": "c-info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              test application
               Instances
            </h1>,
            "className": "c-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "test application",
                " Instances",
              ],
            },
            "ref": null,
            "rendered": Array [
              "test application",
              " Instances",
            ],
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New Application Instance",
            "className": "c-btn c-btn--yellow",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "New Application Instance",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
